### ISSUE TITLE
node.forge and crypto

### ISSUE THREAD

**Issue by richardschneider** [libp2p/js-libp2p-keychain#7]
[node-forge](https://github.com/digitalbazaar/forge) is being used because it supports PKCS 7 and 8.  One major drawback is that it only uses RSA private keys.

`libp2p-crypto` should be enhanced to support PKCS 8 (encrypted private keys).

**Comment by diasdavid** [comment]
Yes, please always augment libp2p-crypto first rather than importing other crypto libraries.

**Comment by richardschneider** [comment]
@diasdavid Rolling our own crypto is not recommened.  libp2p-crypto should be a wrapper around crypto primitives.  That said, PKCS 7 and 8 are not primitives and I agree with you.

**Comment by diasdavid** [comment]
>  libp2p-crypto should be a wrapper around crypto primitives. 

It is what it is currently and I'm just encouraging to continue to be just that.

**Comment by richardschneider** [comment]
Keychain needs the following from a `crypto` library

- PKCS 1, RSA key generation
- PKCS 5, PBKDF2 (password stretcher)
- PKCS 7, Create/Read CMS data
- PKCS 8, Encryption/Decryption of private key with password (pem)
- X.509, Create certificate for a key

**Comment by richardschneider** [comment]
@diasdavid I've searched npmjs and only found two packages that support PKCS 8 with a password; [jsrsasign](https://www.npmjs.com/package/jsrsasign) and [node-forge](https://www.npmjs.com/package/node-forge)

**Comment by richardschneider** [comment]
`libp2p-crypto` now supports PBKDF2.
